Add FireworksAI Embeddings

RobertoBarrosoLuque opened the pull request:
## Summary

Adds FireworksAI integration to the Elasticsearch inference plugin, supporting text embeddings.

This implementation follows the pattern established by other inference services (OpenAI, Cohere, VoyageAI, etc.) and aligns with the structure of PR #134933 (ContextualAI).

## Implementation

**Text Embeddings:**
- OpenAI-compatible API format (`/v1/embeddings`)
- Variable-length embeddings via `dimensions` parameter
- Chunking support for large documents
- Multiple model support (Qwen3 family, Nomic, BERT-based, LLM-based)

## API Reference

Based on [Fireworks AI Embeddings Documentation](https://docs.fireworks.ai/guides/querying-embeddings-models)

---

## Files (17 total)

### Main Implementation (13 files)
```
x-pack/plugin/inference/src/main/java/.../fireworksai/
├── FireworksAiService.java                        # Main service class
├── FireworksAiModel.java                          # Base model class
├── FireworksAiRateLimitServiceSettings.java       # Rate limiting interface
├── FireworksAiResponseHandler.java                # HTTP response handler
├── action/
│   ├── FireworksAiActionCreator.java
│   └── FireworksAiActionVisitor.java
├── embeddings/
│   ├── FireworksAiEmbeddingsModel.java
│   ├── FireworksAiEmbeddingsServiceSettings.java
│   └── FireworksAiEmbeddingsTaskSettings.java
├── request/
│   ├── FireworksAiEmbeddingsRequest.java
│   └── FireworksAiEmbeddingsRequestEntity.java
└── response/
    ├── FireworksAiEmbeddingsResponseEntity.java
    └── FireworksAiErrorResponseEntity.java
```

### Tests (4 files)
- `FireworksAiServiceTests.java`
- `FireworksAiResponseHandlerTests.java`
- `FireworksAiEmbeddingsTaskSettingsTests.java`
- `FireworksAiErrorResponseEntityTests.java`

---

## Usage Example

```bash
# Create an embeddings inference endpoint
PUT _inference/text_embedding/my-fireworks-embeddings
{
  "service": "fireworksai",
  "service_settings": {
    "api_key": "your-api-key",
    "model_id": "fireworks/qwen3-embedding-8b"
  }
}

# Generate embeddings
POST _inference/text_embedding/my-fireworks-embeddings
{
  "input": ["Hello world", "Machine learning is fun"]
}

# With custom dimensions (variable-length embeddings)
POST _inference/text_embedding/my-fireworks-embeddings
{
  "input": ["Hello world"],
  "task_settings": {
    "dimensions": 512
  }
}
```

---

## Configuration Options

### Service Settings
| Setting | Required | Description |
|---------|----------|-------------|
| `api_key` | Yes | FireworksAI API key |
| `model_id` | Yes | Model identifier (e.g., `fireworks/qwen3-embedding-8b`) |
| `url` | No | Custom API endpoint |
| `dimensions` | No | Output embedding dimensions |
| `similarity` | No | Similarity measure (cosine, dot_product, l2_norm) |
| `max_input_tokens` | No | Maximum input tokens |
| `rate_limit` | No | Requests per minute (default: 6000) |

### Task Settings
| Setting | Description |
|---------|-------------|
| `dimensions` | Override dimensions at inference time |

---

## Supported Models

- **Qwen3 Embeddings:** `fireworks/qwen3-embedding-8b`, `4b`, `0p6b`
- **LLM-based:** Any LLM in the Fireworks model library

---

## Checklist

- [x] Follows existing service patterns (similar to ContextualAI PR #134933)
- [x] All precommit checks pass
- [x] Unit tests included
- [x] Transport version registered
- [x] Plugin registration updated
- [x] Integration test updated (`InferenceGetServicesIT`)

---

## Notes

- Simplified to embeddings-only based on reviewer feedback
- Rerank support was removed to focus on core embeddings functionality
- Rate limit default of 6000 RPM per FireworksAI documentation

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,339 @@
+        TimeValue timeout,
+        ActionListener<List<ChunkedInference>> listener
+    ) {
+        if (model.getTaskType() == TaskType.RERANK) {
```
Related to the comment in the FireworksAiEmbeddingsModel. Do we need this check if the FireworksAiEmbeddingsModel check exists just below this and will confirm that the task type is valid?

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,339 @@
+                configurationMap.put(
+                    MODEL_ID,
+                    new SettingsConfiguration.Builder(SUPPORTED_TASK_TYPES).setDescription(
+                        "The model ID to use for FireworksAI requests. Supports Qwen3 embeddings, Nomic embeddings, and reranker models."
```
Do we need to list the specific models supported here? Is it only a subset of the available models for text embedding/rerank in FireworksAI? 

If users can use all text embedding/rerank models available to them, I'd prefer to leave this as just "The model ID to use for FireworksAI requests." as we don't upkeep the available model lists for other services.

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsServiceSettings.java:
```diff
@@ -0,0 +1,283 @@
+        return new FireworksAiEmbeddingsServiceSettings(commonFields, commonFields.dimensions != null);
+    }
+
+    private static CommonFields fromMap(
```
Do these values need to be wrapped in a `CommonFields` record we define? We aren't reusing this record anywhere so I'm not sure why we need the record to encapsulate them?

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsTaskSettings.java:
```diff
@@ -0,0 +1,138 @@
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.V_8_15_0;
```
Should this be set to the FIREWORKS_AI_SERVICE transport version?

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/rerank/FireworksAiRerankTaskSettings.java:
```diff
@@ -0,0 +1,142 @@
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.V_8_15_0;
```
Should this be set to the FIREWORKS_AI_SERVICE transport version?

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsTaskSettings.java:
```diff
@@ -0,0 +1,138 @@
+
+        ValidationException validationException = new ValidationException();
+
+        Integer dimensions = extractOptionalPositiveInteger(map, DIMENSIONS, ModelConfigurations.TASK_SETTINGS, validationException);
```
Dimensions are already tracked within service settings. Why do we also need them in task settings?

RobertoBarrosoLuque replied:
FireworksAI supports variable-length embeddings per the Fireworks AI [documentation](https://docs.fireworks.ai/guides/querying-embeddings-models). The dimensions parameter can be specified per-request to generate embeddings of different sizes.
>
> - service_settings.dimensions: Default dimensions for the model (auto-detected or user-specified at endpoint creation)
> - task_settings.dimensions: Per-request override for variable-length embeddings
>
This allows users to use the same inference endpoint but request different embedding sizes for different use cases (e.g., smaller embeddings for quick similarity checks, larger for precision). If the team prefers simplification, I can remove this from task_settings - just let me know.

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/rerank/FireworksAiRerankServiceSettings.java:
```diff
@@ -0,0 +1,156 @@
+    public FireworksAiRerankServiceSettings(StreamInput in) throws IOException {
+        this.uri = ServiceUtils.createUri(in.readString());
+        this.modelId = in.readString();
+        this.rateLimitSettings = Objects.requireNonNullElse(in.readOptionalWriteable(RateLimitSettings::new), DEFAULT_RATE_LIMIT_SETTINGS);
```
Same comment as in the embeddings service settings, we can use the `new RateLimitSettings(in)` function here.

dan-rubinstein commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiResponseHandlerTests.java:
```diff
@@ -0,0 +1,121 @@
+        MatcherAssert.assertThat(((ElasticsearchStatusException) exception.getCause()).status(), is(RestStatus.BAD_REQUEST));
+    }
+
+    public void testCheckForFailureStatusCode_ThrowsFor429() {
```
Nit: Can we add `_WithShouldRetryTrue` for this test name to be consistent with the others?

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/request/FireworksAiRerankRequest.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public String getInferenceEntityId() {
+        return model != null ? model.getInferenceEntityId() : "unknown";
```
In the constructor above we require that the model is not null. What is the case that it would be null here? Same question applies for the `getURI` function in this class.

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/request/FireworksAiRerankRequest.java:
```diff
@@ -0,0 +1,105 @@
+        return model != null ? model.uri() : null;
+    }
+
+    public Integer getTopN() {
```
Do we need this function? Shouldn't all of the request related classes already have access to this information from the model being passed around?

dan-rubinstein reviewed (commented):
Just a few other questions regarding testing here:
1. What is the best way to test this locally? Is there a free trial or an API key that can be provided for us to test this integration?
2. Can we add unit tests for each of the files that were added that are missing them?
3. Can we ensure that the files that did get unit tests are fully testing all of their public functions (ex. FireworksAiServiceTests should test the parsing/inference functions)?

RobertoBarrosoLuque commented:
> Just a few other questions regarding testing here:
> 
> 1. What is the best way to test this locally? Is there a free trial or an API key that can be provided for us to test this integration?
> 2. Can we add unit tests for each of the files that were added that are missing them?
> 3. Can we ensure that the files that did get unit tests are fully testing all of their public functions (ex. FireworksAiServiceTests should test the parsing/inference functions)?

@dan-rubinstein you can create a test account in the [signup page](https://app.fireworks.ai/signup) I am happy to add some credits once you have it.

I also built elastic internally and tested with API key successfully

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,360 @@
+        // Using 1 token ≈ 0.75 words as a rough estimate: 8000 tokens ≈ 6000 words.
+        // Setting window size to 5500 words to allow headroom for document processing and API overhead.
+        // This conservative estimate ensures compatibility across different rerank model variants.
+        return 5500;
```
Looking at the [model library documentation](https://fireworks.ai/models?modelTypes=Reranker) it seems that all the available models have significantly higher token counts than the Jina rerank of 8000. Is there a way to use Jina models as well through fireworks AI that I am not seeing on this page? Should the limits be different for these models/jina?

dan-rubinstein commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/action/FireworksAiActionCreator.java:
```diff
@@ -0,0 +1,97 @@
+
+    private static final ResponseHandler EMBEDDINGS_HANDLER = new FireworksAiResponseHandler(
+        "fireworksai embeddings",
+        new FireworksAiEmbeddingsResponseEntity(),
```
Should this be calling the `fromResponse` function?

dan-rubinstein reviewed (commented):
I've left a few follow-up comments and resolved any of the comments that were addressed. There's still some remaining comments from the original review that are left unresolved so please go through and let me know if you have any questions about them. I've also created a trial account for FireworksAI. Please let me know what information I need to provide in order for you to provide some credits for testing.

RobertoBarrosoLuque commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/response/FireworksAiErrorResponseEntity.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    public static ErrorResponse fromResponse(HttpResult response) {
+        // Simple error handling - return the status line as the error message
+        // FireworksAI may return error details in the response body, but for now
```
Fixed - Removed the custom class and now using ErrorMessageResponseEntity::fromResponse directly, following the OpenAI pattern.

jonathan-buttner commented on docs/changelog/137130.yaml:
```diff
@@ -0,0 +1,5 @@
+pr: 137130
+summary: Add FireworksAI inference service for embeddings
+area: Machine Learning
```
```suggestion
area: Inference
```

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsServiceSettings.java:
```diff
@@ -0,0 +1,256 @@
+        );
+
+        // Optimize URI conversion - only convert if URL is provided, otherwise use default
+        URI uri = url != null
```
Can you point me to an example in the firework docs of where a user can specify a url to a deployment? In most of the examples I see references to: `https://api.fireworks.ai/inference/v1/embeddings`

Would a user ever use a custom url?

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsServiceSettings.java:
```diff
@@ -0,0 +1,256 @@
+        this.modelId = in.readString();
+        this.uri = ServiceUtils.createUri(in.readString());
+        this.similarity = in.readOptionalEnum(SimilarityMeasure.class);
+        this.dimensions = in.readOptionalInt();
```
Let's use the VInt variant

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsServiceSettings.java#L186-L187

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,331 @@
+        Map<String, Object> taskSettingsMap = removeFromMapOrDefaultEmpty(config, ModelConfigurations.TASK_SETTINGS);
+        Map<String, Object> secretSettingsMap = removeFromMapOrDefaultEmpty(secrets, ModelSecrets.SECRET_SETTINGS);
+
+        ChunkingSettings chunkingSettings = ChunkingSettingsBuilder.fromMap(removeFromMap(config, ModelConfigurations.CHUNKING_SETTINGS));
```
I realize that text embedding is the only task type supported but how about we wrap this in an if block check the task type. That way we don't get failures in the future if/when we extend the implementation to other task types.

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/OpenAiService.java#L240-L242

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,331 @@
+            Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.SERVICE_SETTINGS);
+            Map<String, Object> taskSettingsMap = removeFromMapOrDefaultEmpty(config, ModelConfigurations.TASK_SETTINGS);
+
+            ChunkingSettings chunkingSettings = ChunkingSettingsBuilder.fromMap(
```
Same for here, let's wrap in an if-block

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,339 @@
+            return configuration.getOrCompute();
+        }
+
+        private static final LazyInitializable<InferenceServiceConfiguration, RuntimeException> configuration = new LazyInitializable<>(
```
@RobertoBarrosoLuque yeah how about we add `dimensions` as well?

Depending on the response about the url, we may want to add that too

Example: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/OpenAiService.java#L459-L471

jonathan-buttner reviewed (changes requested):
Looking good! Thanks for the changes, I left a few more suggestions and questions.

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/request/FireworksAiEmbeddingsRequest.java:
```diff
@@ -41,8 +41,10 @@ public FireworksAiEmbeddingsRequest(List<String> input, FireworksAiEmbeddingsMod
+        // Only include dimensions in the request if explicitly set by the user.
+        // Some models don't support the dimensions parameter, so we only send it when user configured it.
+        Integer dimensions = null;
+        if (Boolean.TRUE.equals(model.getServiceSettings().dimensionsSetByUser())) {
```
I think we want the check to be like this:

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/request/OpenAiEmbeddingsRequestEntity.java#L46-L48

jonathan-buttner commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsTaskSettingsTests.java:
```diff
@@ -7,97 +7,56 @@
 
```
I think we can remove this file if we remove the implementation.

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiModel.java:
```diff
@@ -0,0 +1,84 @@
+        return apiKey().hashCode();
+    }
+
+    public URI baseUri() {
```
I don't think this is used, can we remove it?

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiModel.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    public int rateLimitGroupingHash() {
+        return apiKey().hashCode();
```
Is the rate limit only tied to the api key? Or is the URL and model name used as well?

jonathan-buttner commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiServiceTests.java:
```diff
@@ -0,0 +1,73 @@
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.mock;
+
+public class FireworksAiServiceTests extends InferenceServiceTestCase {
```
Let's follow `OpenAiServiceTests` as an example for how to test the service.

jonathan-buttner reviewed (commented):
Thanks for the changes.

A few more things:

- Let's add the named writeables to `InferenceNamedWriteablesProvider`
- Add tests
  - Take a look at how we test other integrations, `OpenAI` is a pretty good example
  - Add tests for the service settings file
  - Add tests for `FireworksAiEmbeddingsRequestEntity`
  - Add a class similar to `OpenAiServiceParameterizedTests`
  - Add a class similar to `OpenAiServiceTests`
    - Make sure we test end to end like the `testInfer_SendsRequest`
    - Take a look through the other tests in this class, a number of them won't apply (like the stuff around streaming, completion, etc) but a lot of the tests will
    -

jonathan-buttner commented on server/src/main/resources/transport/upper_bounds/9.4.csv:
```diff
@@ -1 +1,2 @@
+fireworks_ai_service,9257000
```
Hmm this file should only have a single entry in it. Let's revert this change.

When resolving conflicts the flow is to merge in main and then run this command:

```
./gradlew resolveTransportVersionConflict
```

jonathan-buttner commented on x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiService.java:
```diff
@@ -0,0 +1,352 @@
+ * FireworksAI inference service for text embeddings.
+ * This service uses the FireworksAI REST API to perform text embeddings.
+ */
+public class FireworksAiService extends SenderService {
```
I think we need to merge in main because we've added some new methods that need to be implemented for each service `buildModelFromConfigAndSecrets`.

nithiyn replied:
merged in main, this introduced the abstract buildModelFromConfigAndSecrets method. Implemented it using the `ModelCreator` pattern: created `FireworksAiEmbeddingsModelCreator`, added a `MODEL_CREATORS` map, and used `retrieveModelCreatorFromMapOrThrow()` to dispatch. Also refactored `createModel()` to use the same pattern, and added a (ModelConfigurations, ModelSecrets) constructor to FireworksAiEmbeddingsModel.

jonathan-buttner commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiServiceTests.java:
```diff
@@ -0,0 +1,178 @@
+
+    @Override
+    public void testParseRequestConfig_CreatesACompletionModel() {
+        // FireworksAI does not support the completion task type
```
Ah, I'll see if we can modify the framework to we don't have to override it like this in the future.

jonathan-buttner commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/FireworksAiServiceTests.java:
```diff
@@ -0,0 +1,178 @@
+/*
```
Can you add a test similar to the `OpenAiServiceTests::testInfer_SendsRequest`?

Let's also add a test file similar to `OpenAiServiceParameterizedParsingTests` and `OpenAiServiceParameterizedModelCreationTests`.

jonathan-buttner reviewed (changes requested):
Looking good!

Could you add tests for:

- `FireworksAiEmbeddingsRequestEntity`
- `FireworksAiEmbeddingsServiceSettings`
- A test class just like `OpenAiServiceParameterizedParsingTests`
- a test class just like `OpenAiServiceParameterizedModelCreationTests`

nithiyn committed:
Add FireworksAI inference service for text embeddings

Adds a new inference service integration for FireworksAI supporting
text embeddings via the FireworksAI REST API.

Includes service implementation, embeddings model, request handling,
action creators, and comprehensive test coverage including wire
serialization, parameterized parsing, model creation, and integration tests.

nithiyn commented:
> Looking good!
> 
> Could you add tests for:
> 
> * `FireworksAiEmbeddingsRequestEntity`
> * `FireworksAiEmbeddingsServiceSettings`
> * A test class just like `OpenAiServiceParameterizedParsingTests`
> * a test class just like `OpenAiServiceParameterizedModelCreationTests`

thank you for the quick turnaround on the feedback @jonathan-buttner! please lmk how the latest looks

jonathan-buttner commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/embeddings/FireworksAiEmbeddingsServiceSettingsTests.java:
```diff
@@ -0,0 +1,524 @@
+        entity.toXContent(builder, null);
+        String xContentResult = Strings.toString(builder);
+
+        assertThat(xContentResult, is("""
```
Let's use `stripWhitespace` to make the json a little more readable:

For example we can have something like this
```
var expected = XContentHelper.stripWhitespace("""
            {
                "model_id": "some model",
                "rate_limit": {
                    "requests_per_minute": 2
                }
            }
            """);
```

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/ai21/completion/Ai21ChatCompletionServiceSettingsTests.java#L94-L103

Let's use that helper for all the tests in this class that handle json.

jonathan-buttner commented on x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/fireworksai/request/FireworksAiEmbeddingsRequestEntityTests.java:
```diff
@@ -0,0 +1,74 @@
+        entity.toXContent(builder, null);
+        String xContentResult = Strings.toString(builder);
+
+        assertThat(xContentResult, is("""
```
Let's use the helper `stripWhitespace` for the json in this file too

jonathan-buttner reviewed (commented):
Looks good, just a few minor points. Thanks for the changes!

jonathan-buttner commented:
@elasticmachine test this please

nithiyn committed:
Address review feedback: formatting, stripWhitespace, and docs link

- Fix checkstyle line length violation in FireworksAiService (>140 chars)
- Use XContentHelper.stripWhitespace() for JSON assertions in
  FireworksAiEmbeddingsServiceSettingsTests and
  FireworksAiEmbeddingsRequestEntityTests for readability
- Replace custom TIMEOUT with inherited ESTestCase.TEST_REQUEST_TIMEOUT
  in FireworksAiServiceTests
- Add FireworksAI embeddings models documentation link to dimensions
  Configuration description

nithiyn committed:
Merge remote-tracking branch 'upstream/main'

# Conflicts:
#	server/src/main/resources/transport/upper_bounds/9.4.csv
#	x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceNamedWriteablesProvider.java

nithiyn committed:
Resolve transport version conflict after merging upstream main

nithiyn committed:
Merge remote-tracking branch 'upstream/main'

nithiyn committed:
Merge remote-tracking branch 'upstream/main'

Resolve transport version conflict after merging upstream main

nithiyn committed:
Merge branch 'main' into main

jonathan-buttner committed:
Merge branch 'main' into main

nithiyn committed:
Merge branch 'main' into main

jonathan-buttner committed:
Merge branch 'main' of github.com:elastic/elasticsearch into RobertoBarrosoLuque/main
